src/downshift.js

```diff
@@ -952,6 +954,11 @@ class Downshift extends Component {
   }
 }
 
+// Polyfill your component so the new lifecycles will work with older versions of React:
+if (Component.prototype && Component.prototype.isReactComponent) {
+  polyfill(Downshift)
```

Comment:
I'm pretty sure that we'll need to reassign Downshift to this for this to work properly:

```js
Downshift = polyfill(Downshift)
```

Perhaps we should add a test to make sure this polyfill is applied (not certain how to do that though... maybe it's not worth it).

Reply:
Just kidding. I just checked and you've got this right. Thanks! We probably still need to actually use the new lifecycle methods though...